### DISCUSSION TITLE
How to trigger hide sub-notes?

### DISCUSSION THREAD

**Discussion by Nriver** [TriliumNext/Trilium#3411] [Q&A] [answered]
I'm using ETAPI to upload markdown files that contains images.

https://github.com/Nriver/trilium-py/blob/main/src/trilium_py/client.py#L546

Here is what I did:
1. create a empty note
2. upload images as sub-notes
3. replace images url in note to Trilium standard ones
4. update the note content

So far, the note contents are good, images are shown properly.

However, the image sub-notes are not hidden as a normal note would do. I had to manually open the note, then modify it a bit, like enter a new line, then the sub-notes will go hidden.

![1](https://user-images.githubusercontent.com/6752679/207278839-62732ede-8f6a-45cc-82e3-94361b881392.gif)

How can I trigger this process with code? It would be nice if I can trigger this in ETAPI.

**Comment by zadam** [comment, marked as answer]
Hi, Trilium has a "system" relation `~imageLink` in the direction from the text note pointing to the image note.

This is used to decide whether the children should be shown or not.

When you do the little edit, the note is parsed and the imageLink relations generated.

To avoid having to do this, simply create these `~imageLink` relations using ETAPI.

**Reply by Nriver** [reply]
Nailed it! Thank you!
